app/services/mentions/create_all.rb

```diff
@@ -10,32 +10,55 @@ def self.call(*args)
 
     def call
       # Only works for comments right now.
+      mentioned_users = users_mentioned_in_notifiable_text_that_are_existing_and_not_its_owners
+
+      delete_mentions_removed_from_notifiable_text(mentioned_users)
+      create_mentions_for(mentioned_users)
+    end
+
+    private
+
+    def users_mentioned_in_notifiable_text_that_are_existing_and_not_its_owners
+      mentioned_usernames = extract_usernames_from_mentions_in_text
+
+      collect_existing_users(mentioned_usernames).
+        then do |existing_mentioned_users|
+          reject_notifiable_owner existing_mentioned_users
+        end
+    end
+
+    def collect_existing_users(usernames)
+      usernames.filter_map do |username|
+        User.find_by(username: username)
+      end
+    end
+
+    def create_mentions_for(users)
+      users.map { |user| create_mention_for(user) }
+    end
+
+    def extract_usernames_from_mentions_in_text
       # Paired with the process that creates the "comment-mentioned-user"
       doc = Nokogiri::HTML(notifiable.processed_html)
-      usernames = []
-      mentions = []
-      doc.css(".comment-mentioned-user").each do |link|
-        username = link.text.delete("@").downcase
-        user = User.find_by(username: username)
-
-        if user && user.id != notifiable.user_id
-          usernames << username
-          mentions << create_mention(user)
-        end
+      doc.css(".comment-mentioned-user").map do |link|
+        link.text.delete("@").downcase
       end
-      delete_removed_mentions(usernames)
-      mentions
     end
 
-    private
+    def reject_notifiable_owner(users)
+      users.reject { |user| user_owns_notifiable?(user, @notifiable) }
+    end
+
+    def user_owns_notifiable?(user, notifiable)
+      user.id == notifiable.user_id
+    end
```

Comment:
It seems to me that this doesn't really need to be a method since it's only ever used in `reject_notifiable_owner`'s `reject` block; we could probably just do it inline there.

Reply:
I strongly disagree. It's easier to understand and `authored_by?(user, notifiable)` (improved on the previous name) is revealing a domain concept.